Library/Homebrew/formula.rb

```diff
@@ -2038,11 +2057,16 @@ def needs(*standards)
     #
     # The test will fail if it returns false, or if an exception is raised.
     # Failed assertions and failed `system` commands will raise exceptions.
-
     def test(&block)
       define_method(:test, &block)
     end
 
+    # TODO
+    def pour_bottle?(reason, &block)
+      @pour_bottle_reason = reason
+      @pour_bottle_block = block
```

Comment:
> We should allow no reason. The current pour_bottle? nil looks weird.

Agreed it's weird. I'd like to us avoid having no reason as it's user-hostile but I've thought of the right reason for this case.




> Instead of introducing new pour_bottle_block variable, we can store it in pour_bottle? as define_method

Cool idea, will try it out.